### ISSUE TITLE
Convert ArchiveEditDialog to modal from nested page element

### ISSUE THREAD

**Issue by AdamJ** [AdamJ/TimeTrackerPro#71]
The current `ArchiveEditDialog` renders as a card in the inline scroll area, which the user can easily scroll past. There is no focus management either.

## ToDo

- [ ] Replace with an `AlertDialog` nested inside of the existing dialog, triggered by the Delete button
